### PR TITLE
Add video support

### LINKED ISSUE #58: video support
Currently, when you embed a video, then you need to click on the link to see the video.

A better UX workflow can place the video embed after the first click.

I need to review this behavior in order to determine if it could be possible on the front end, also keeping in mind bundle size for the extra code for that.

A good approximation to this could be do something similar to how YouTube video preview works.

Metadata example:

- https://api.microlink.io/?url=https://twitter.com/verge/status/957383241714970624
- https://plays.tv/video/5a6f64b1bef69a7fa9/holy-shit
- https://gfycat.com/gifs/detail/timelyhealthyarmadillo

### PR DESCRIPTION
**Pull request by Kikobeats**
Closes #58 

**New video API**

- [x] Make API configurable from interface
  - [x] autoPlay
  - [x] muted
  - [x] loop
- [x] Add video support documentation.
- [x] Add CSS markup documentation.

### CODE REVIEW

File: packages/react-microlink/src/components/Card/CardImage/image.js [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+import { media, isLarge } from '../../../utils'
+
+const largeStyle = css`
+flex: 1;
+
+&::before {
+  padding-bottom: 0;
+}
+
+${media.mobile`
```
[the remark this answers is not in the document]
Comment (Kikobeats) [reply packages/react-microlink/src/components/Card/CardImage/image.js]: good catch 👌

File: packages/react-microlink/src/components/Card/CardMedia/video.js [hunk 2]
```diff
@@ -39,16 +39,23 @@ const Video = styled.video`
...
-const CardVideo = ({cardSize, video, image, muted, autoPlay, loop}) => {
+const CardVideo = ({cardSize, video, image, muted, autoplay, loop}) => {
   return (
     <VideoWrapper cardSize={cardSize}>
       <Video
         src={video}
         poster={image}
         muted={muted}
-        autoPlay={autoPlay}
+        autoplay={autoplay}
```
Comment (Kikobeats) [inline packages/react-microlink/src/components/Card/CardMedia/video.js]: I think react says need to be `autoPlay`

Comment (breadadams) [reply packages/react-microlink/src/components/Card/CardMedia/video.js]: Whoops of course, camelCase props 🤦‍♂️

Comment (breadadams) [reply packages/react-microlink/src/components/Card/CardMedia/video.js]: Weirdly, `playsinline` works 🤔

File: packages/react-microlink/src/components/Card/CardMedia/video.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: 100;
+
+  ${({autoPlay}) => autoPlay && `
+    &::-webkit-media-controls-start-playback-button {
+      display: none!important;
+      -webkit-appearance: none;
```
Comment (Kikobeats) [inline packages/react-microlink/src/components/Card/CardMedia/video.js]: what's that 🤔

Comment (breadadams) [reply packages/react-microlink/src/components/Card/CardMedia/video.js]: It hides the typical ▶️ button from the `<video />` element, ie. this:

image

It was displaying over the videos even if `autoplay` was enabled, so I've hidden it via CSS when `autoPlay === true`

**Comment by breadadams** [comment]
Now that `videoAutoplay={false}` is a thing, I guess a _play_ button would be necessary @Kikobeats

File: packages/react-microlink/src/components/Card/CardMedia/video.js (hunk 3 above)
Comment (Kikobeats) [reply packages/react-microlink/src/components/Card/CardMedia/video.js]: 👌

**Pull request merged**